cns/cnireconciler/statefile.go

```diff
@@ -0,0 +1,38 @@
+package cnireconciler
+
+import (
+	"encoding/json"
+	"errors"
+	"os"
+)
+
+// WriteObjectToCNIStatefile checks for a file at the CNI statefile path,
+// and checks if it is empty. If it is empty, writes an empty JSON object to
+// it so older CNI can execute. Does nothing and returns no error if the
+// file does not exist.
+//
+// This is a hack to get older CNI to run when CNS has mounted the statefile
+// path, but the statefile wasn't written by CNI yet. Kubelet will stub an
+// empty file on the host filesystem, crashing older CNI because it doesn't know
+// how to handle empty statefiles.
+func WriteObjectToCNIStatefile() error {
+	filename := "/var/run/azure-vnet.json"
+	return writeObjectToFile(filename)
+}
+
+func writeObjectToFile(filename string) error {
+	fi, err := os.Stat(filename)
+	if err != nil {
+		if errors.Is(os.ErrNotExist, err) {
+			return nil
+		}
+		return err
+	}
+
+	if fi.Size() != 0 {
+		return nil
+	}
+
+	b, _ := json.Marshal(map[string]string{})
```

Comment:
Shouldnt we acquire the lock? Can there be a race where it will accidently update the correct file?

Reply:
can you also validate reconcile works fine, say cns crashes write after updating this

Reply:
i think that's not possible. I guess this code is at start of cns. cni dependent on cns to get ip and create endpoint. cni would be in error state unless it get ip from cns

Reply:
This is a race, 

T1. CNS starts up file
T2.  CNI gets an IP but hasnt updated the file yet
T3. CNS crashes again
T4. CNS starts checks File size is still 0
T5. After CNS file size check, CNI updates the IP it got at T2
T6. CNS overwrites with empty json

Seems a very rare scenario but it could still happen

Reply:
we can't acquire the lock because in old versions of CNI that path is just `/var/run/azure-vnet.lock` and we're back at the same problem with the hostpath mount

Reply:
@neaggarwMS cns should not return ip to cni unless it creates empty json file. so T4 won't occur.

Reply:
@tamilmani1989, it will during T1 it already created an empty json and became active.

Reply:
if CNS has written `{}` to the file, the file size will be > 0, and it won't re-write `{}` to the file

Reply:
then file size won't be 0 at T4

Reply:
ah, makes sense.